ocw/lib/eks.py

```diff
@@ -105,40 +105,49 @@ def wait_for_empty_nodegroup_list(self, region: str, cluster_name: str, timeout_
         if self.dry_run:
             self.log_info("Skip waiting due to dry-run mode")
             return None
-        self.log_dbg("Waiting empty nodegroup list in {}", cluster_name)
+        self.log_dbg(f"Waiting empty nodegroup list in {cluster_name}")
         end = datetime.now(timezone.utc) + timedelta(minutes=timeout_minutes)
         resp_nodegroup = self.eks_client(region).list_nodegroups(clusterName=cluster_name)
 
         while datetime.now(timezone.utc) < end and len(resp_nodegroup['nodegroups']) > 0:
             time.sleep(20)
             resp_nodegroup = self.eks_client(region).list_nodegroups(clusterName=cluster_name)
             if len(resp_nodegroup['nodegroups']) > 0:
-                self.log_dbg("Still waiting for {} nodegroups to disappear", len(resp_nodegroup['nodegroups']))
+                self.log_dbg(f"Still waiting for {len(resp_nodegroup['nodegroups'])} nodegroups to disappear")
         return None
 
     def delete_all_clusters(self) -> None:
         self.log_info("Deleting all clusters!")
         for region in self.__cluster_regions:
             response = self.eks_client(region).list_clusters()
             if len(response['clusters']):
-                self.log_dbg("Found {} cluster(s) in {}", len(response['clusters']), region)
-                for cluster in response['clusters']:
-                    resp_nodegroup = self.eks_client(region).list_nodegroups(clusterName=cluster)
-                    if len(resp_nodegroup['nodegroups']):
-                        self.log_dbg("Found {} nodegroups for {}", len(resp_nodegroup['nodegroups']), cluster)
-                        for nodegroup in resp_nodegroup['nodegroups']:
-                            if self.dry_run:
-                                self.log_info("Skipping {} nodegroup deletion due to dry-run mode", nodegroup)
-                            else:
-                                self.log_info("Deleting {}", nodegroup)
-                                self.eks_client(region).delete_nodegroup(
-                                    clusterName=cluster, nodegroupName=nodegroup)
-                        self.wait_for_empty_nodegroup_list(region, cluster)
+                self.log_dbg(f"Found {len(response['clusters'])} cluster(s) in {region}")
+            for cluster in response['clusters']:
+                resp_nodegroup = self.eks_client(region).list_nodegroups(clusterName=cluster)
+                if len(resp_nodegroup['nodegroups']):
```

Comment:
```suggestion
                if 'nodegroups' in resp_nodegroup:
```
I think it's better to check for the key in the dict, otherwise we get a `KeyError` if the item is not present.